elytron/src/test/java/org/wildfly/extension/elytron/KeyStoresTestCase.java

```diff
@@ -96,8 +114,197 @@ private ModelNode assertFailed(ModelNode response) {
+
+        X509Certificate rootCertificate = rootSelfSignedX509CertificateAndSigningKey.getSelfSignedCertificate();
+        testKeyStore.setKeyEntry("ca", rootSelfSignedX509CertificateAndSigningKey.getSigningKey(), KEY_PASSWORD.toCharArray(), new X509Certificate[]{rootCertificate});
+
+        return testKeyStore;
+    }
+
+    private static X509Certificate createTestTrustedCertificate(SelfSignedX509CertificateAndSigningKey rootSelfSignedX509CertificateAndSigningKey) throws Exception {
+        KeyPairGenerator keyPairGenerator = KeyPairGenerator.getInstance("RSA");
+        KeyPair keys = keyPairGenerator.generateKeyPair();
+        PublicKey publicKey = keys.getPublic();
+
+        return new X509CertificateBuilder()
+                .setIssuerDn(ROOT_DN)
+                .setSubjectDn(INTERMEDIATE_DN)
+                .setSignatureAlgorithmName("SHA1withDSA")
+                .setSigningKey(rootSelfSignedX509CertificateAndSigningKey.getSigningKey())
+                .setPublicKey(publicKey)
+                .build();
+    }
+
+    private static X509Certificate createTestUntrustedCertChainReplyCertificate(SelfSignedX509CertificateAndSigningKey selfSignedX509CertificateAndSigningKey, PublicKey publicKey) throws Exception {
+        return new X509CertificateBuilder()
+                .setIssuerDn(ANOTHER_ROOT_DN)
+                .setSubjectDn(SSMITH_DN)
+                .setSignatureAlgorithmName("SHA1withDSA")
+                .setSigningKey(selfSignedX509CertificateAndSigningKey.getSigningKey())
+                .setPublicKey(publicKey)
+                .build();
+    }
+
+    private static void setUpFiles() throws Exception {
+        File workingDir = new File(WORKING_DIRECTORY_LOCATION);
+        if (workingDir.exists() == false) {
+            workingDir.mkdirs();
+        }
+
+        SelfSignedX509CertificateAndSigningKey rootSelfSignedX509CertificateAndSigningKey = createCertRoot(ROOT_DN);
+        SelfSignedX509CertificateAndSigningKey anotherRootSelfSignedX509CertificateAndSigningKey = createCertRoot(ANOTHER_ROOT_DN);
+        SelfSignedX509CertificateAndSigningKey sallySelfSignedX509CertificateAndSigningKey = createSallyCertificate();
+
+        PublicKey publicKey = sallySelfSignedX509CertificateAndSigningKey.getSelfSignedCertificate().getPublicKey();
+
+        // Key Stores
+        File fireflyFile = new File(workingDir, "firefly.keystore");
+        File testFile = new File(workingDir, "test.keystore");
+
+        KeyStore fireflyKeyStore = createFireflyKeyStore();
+        KeyStore testKeyStore = createTestKeyStore(rootSelfSignedX509CertificateAndSigningKey, sallySelfSignedX509CertificateAndSigningKey);
+
+        createTemporaryKeyStoreFile(fireflyKeyStore, fireflyFile);
+        createTemporaryKeyStoreFile(testKeyStore, testFile);
+
+        // Cert Files
+        File testSingleCertReplyFile = new File(workingDir, "test-single-cert-reply.cert");
```

Comment:
To avoid some duplication, you could create static variables for the file declarations and then reference them in this method and in ```removeTestFiles```.

Reply:
Fixed